### ISSUE TITLE
Pointer events on child actors after moving

### ISSUE THREAD

**Issue by Osthekake** [excaliburjs/Excalibur#1493]
I am having a problem with pointer events on child actors, where the area that triggers them, seem to be based on the original position of the child, rather than their current position.

I have this problem reliably in a game I made for a jam way back that I have recently been trying to refurbish. I am pretty certain that this is a problem with excalibur, but I have not attempted making a smallest possible example, so I cannot be 100% sure. Should I link my original project here?

### Steps to Reproduce
1. Implement drag and drop like this
```
const parent = new Actor();
const child = new Actor();
parent.add(child);
let pointerDown = false;
child.on("pointerdown", () => { 
  pointerDown = true; 
});
child.on("pointerup", () => { 
  pointerDown = false; 
  console.log('pointerup happens');
});
child.on("pointermove", (pe: ex.Input.PointerEvent) => {
  if (!pointerDown) return;
  // something that moves the child actor to where the pointer is
});
```

2. drag and drop the child actor to somewhere other than where it started.

### Expected Result
Console should output "pointerup happens" even when the child actor is moved outside of its original position.

### Actual Result
Console outputs "pointerup happens" only when pointer is inside the original child actor position when it is released.

### Environment
- browsers and versions: Chrome (80.0.3987.163)
- operating system: Windows 10
- Excalibur versions: 0.18 and 0.24 for sure. Haven't tried others, but I imagine it may be a problem in all of them.

### Current Workaround
It's possible to catch the pointerup event using something like this:
```
engine.input.pointers.primary.on('up', (event: ex.Input.PointerEvent) => {
      if (pointerDown) {
              // do stuff
      }
});
```
Unfortunately, this only solves the problem with pointer up. Pointer down will not trigger from the new position either. A similar workaround for that negates the entire point of having the pointer events on the actor.

**Comment by eonarheim** [comment]
@Osthekake if you are still interested, linking the project where you were having the problem would also help

We've made some updates to latest `main` around pointers published in the alpha npm @ next, potentially fixed?
![image](https://user-images.githubusercontent.com/612071/121759697-a4beea80-caec-11eb-9553-6d0fea2347a0.png)

**Comment by Osthekake** [comment]
This is exciting!
The project is this one: https://github.com/Osthekake/ld42 which is actually 3 years old. I've been working on trying to bump it to newest excalibur version over the past weekend, but I didn't manage to finish figuring out the new tiled system, and I didn't find an adequate workaround to force it to build. 

Let me get back to you next week, and I can tell you if it works. :)

Since there is a clear divide between addChild and other adds, that might help fix issues. However, parent is now an entity, which I cannot read pos and rotation of. I don't know if my fix of simply casting them works yet, since my project does not even compile. 
Is there a way to know pos and rotation of your parent, or is that simply something which I should not need if I do things "correctly" ?

**Comment by eonarheim** [comment]
@Osthekake I'm happy to report this appears to be fixed in the latest excalibur release v0.25.1!

```typescript
var game = new ex.Engine({
  width: 600,
  height: 400
});

var parent2 = new ex.Actor({x: 0, y: 0, width: 100, height: 100, color: ex.Color.Blue});
var child = new ex.Actor({x: 10, y: 10, width: 50, height: 50, color: ex.Color.Green});
parent2.addChild(child);

game.add(parent2);

let pointerDown = false;
child.on("pointerdown", () => { 
  pointerDown = true; 
});
child.on("pointerup", () => { 
  pointerDown = false; 
  console.log('pointerup happens');
});
child.on("pointermove", (pe: ex.Input.PointerEvent) => {
  if (!pointerDown) return;
  // something that moves the child actor to where the pointer is
  child.transform.globalPos = pe.worldPos;
});

game.start();
```
